app/javascript/lib/validation/validate-file-content.js

```diff
@@ -144,17 +144,72 @@ function validateEqualCount(headers, annotTypes) {
   return issues
 }
 
+/**
+ * Verify "GENE" is present as the first column in the first row for an Expression Matrix file
+ */
+function validateGeneInHeader(firstRowTable) {
+  const issues = []
+
+  if (firstRowTable[0][0] !== 'GENE') {
+    const msg = 'Dense matrices require the first value of the file to be "GENE". ' +
+     `However, the first value for this file currently is "${firstRowTable[0][0]}".`
+    issues.push(['error', 'format:missing-gene-column', msg])
+  }
+
+  return issues
+}
 
 /**
- * Verify cell names are each unique for a cluster or metadata file
+ * Verify dense matrix column values and column numbers
  */
-function validateUniqueCellNamesWithinFile(table) {
+function validateColumnNumberAndValues(table) {
+  const issues = []
+  const rowsWithIncorrectColumnNumbers = []
+  const rowsWithNonNumericValues = []
+  // use the header row to determine the correct number of columns all rows should have
+  const correctNumberOfColumns = table[0].length
+
+  // skip the first row
+  for (let i = 1; i < table.length - 1; i++) {
+    const entry = table[i]
+    if (correctNumberOfColumns !== entry.length) {
+      rowsWithIncorrectColumnNumbers.push(i)
+    }
+    // skip first column
+    entry.shift()
+    if (entry.some(isNaN)) {
+      rowsWithNonNumericValues.push(i)
+    }
+  }
+
+  if (rowsWithIncorrectColumnNumbers.length > 0) {
+    const rowText = rowsWithIncorrectColumnNumbers.length > 1 ? 'rows' : 'row'
+    const msg = `All rows must have the same number of columns - ` +
+    `please ensure the number of columns for ${rowText}: ${rowsWithIncorrectColumnNumbers.join(', ')}, ` +
+    `matches the file-header-specificed number of ${correctNumberOfColumns} columns-per-row.`
+    issues.push(['error', 'format:mismatch-column-number', msg])
+  }
+
+  if (rowsWithNonNumericValues.length > 0) {
+    const rowText = rowsWithNonNumericValues.length > 1 ? 'rows' : 'row'
+    const msg = `All values (other than the first column and header row) in a dense matrix file must be numeric. ` +
+    `Please ensure all values in ${rowText}: ${rowsWithNonNumericValues.join(', ')}, are numbers.`
+    issues.push(['error', 'format:invalid-type:not-numeric', msg])
+  }
+
+  return issues
+}
+
+/**
+ * Verify cell names are each unique for a file
+ */
+function validateUniqueCellNamesWithinFile(table, fileType) {
   const issues = []
 
   const cellNames = new Set()
   const duplicates = new Set()
   for (let i = 0; i < table.length; i++) {
-    const cell = table[i][0]
+    const cell = fileType === 'Expression Matrix' ? table[i]: table[i][0]
```

Comment:
A space before the colon is conventional in ternaries.

```suggestion
    const cell = fileType === 'Expression Matrix' ? table[i] : table[i][0]
```